### ISSUE TITLE
flask addon can't get post data

### ISSUE THREAD

**Issue by orleven** [mitmproxy/mitmproxy#4717]
#### Problem Description
flask addon can't get post data

#### Steps to reproduce the behavior:
```
# test.py

from flask import Flask, request
from mitmproxy.addons import asgiapp

app = Flask("proxapp")

@app.route('/', methods=["GET", "POST"])
def hello_world():
    text = "request.data: "
    text += str(request.data)
    text += "\r\n"

    text += "request.form: "
    text += str(request.form)
    text += "\r\n"

    text += "request.json: "
    text += str(request.json)
    text += "\r\n"

    text += "request.values: "
    text += str(request.values)
    text += "\r\n"

    text += "request.args: "
    text += str(request.args)
    text += "\r\n"
    return text

addons = [
    asgiapp.WSGIApp(app, "example.com", 80)
]
```

mitmproxy -s test.py   --listen-host 127.0.0.1 -p 8002

and I can't get the post data.

#### System Information
Mitmproxy: 7.0.0
Python:    3.9.6
OpenSSL:   OpenSSL 1.1.1k  25 Mar 2021
Platform:  macOS-10.14.6-x86_64-i386-64bit

**Comment by mhils** [comment]
Thanks for the excellent report! I discovered two issues here:

- `curl -x localhost:8080 -X POST -F "foo=bar" http://example.com -vvv` failing: We incorrectly did not lowercase headers, this will be fixed with #4722
- `curl -x localhost:8080 -X POST -F "foo=bar" http://example.com -vvv -H "transfer-encoding: chunked"` failing: This is an upstream issue with asgiref, I'll submit a PR there shortly and will link here. 😃

**Comment by orleven** [comment]
thank you!  😃
